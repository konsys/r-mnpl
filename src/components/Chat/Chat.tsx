import React from "react";

export const Chat = () => (
  <>
    <div className="table-body-board-chat scr" scr-active="1">
      <div
        className="scr-window"
        style={{ width: "525px", height: "439px", paddingTop: "439px" }}
      >
        <div className="scr-content" style={{ width: "475px" }} />
        <div className="scr-pane" style={{ display: "none" }}>
          <div
            className="scr-pane-handler"
            style={{ transform: "translateY(0px)" }}
          />
        </div>
      </div>
    </div>
    <div className="table-body-board-chatbottom">
      <div className="_channel" mnpl-ch="all" />
      <div className="_input">
        <input placeholder="Введите сообщение" max-lenght={200} />
        <div
          mnpl-emote="Rip"
          className="emotes-button"
          style={{ backgroundImage: `url("//m1.dogecdn.wtf/chat/eRip.svg")` }}
        ></div>
        <div className="_hint _hint_tab">
          Теперь можно писать конкретному игроку!
          <br />
          <span />, чтобы выбрать получателя.
        </div>
      </div>
      <div className="_hidekeyboard" />
      <div className="_screenshot" style={{ display: "none" }} />
      <div className="_fullscreen" />
      <div className="_fullscreenoff" style={{ display: "none" }} />
      <div className="_helper" />
    </div>
  </>
);
