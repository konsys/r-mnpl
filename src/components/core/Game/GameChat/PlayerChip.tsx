import { Chip } from "@material-ui/core";
import FaceIcon from "@material-ui/icons/Face";
import React from "react";
export default function PlayerChip({
  handleDelete,
  name,
}: {
  handleDelete: () => any;
  name: string;
}) {
  return (
    <>
      <Chip
        size="small"
        icon={<FaceIcon />}
        label={name}
        onDelete={handleDelete}
        variant="outlined"
        color="default"
      />
    </>
  );
}
