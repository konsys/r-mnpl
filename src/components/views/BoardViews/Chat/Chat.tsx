import React from "react";

export const Chat = () => (
  <>
    <div className="table-body-board-chat scr" scr-active="1">
      <div
        className="scr-window"
        style={{ paddingTop: "0px", width: "525px", height: "439px" }}
      >
        <div className="scr-content" style={{ width: "475px" }}>
          <div className="table-body-board-chat-message" id="event_CUafwqq8AAM">
            <span
              className="table-body-board-chat-message-user _color_player_0_color _rendered"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            выбрасывает 1:2
          </div>
          <div className="table-body-board-chat-message" id="event_CUafwqs8AAQ">
            <span
              className="table-body-board-chat-message-user _color_player_0_color _rendered"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            попадает на
            <span className="_field" mnpl-field_id="3">
              Hugo Boss
            </span>{" "}
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUafxQq8AAc">
            <span
              className="table-body-board-chat-message-user _color_player_0_color _rendered"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            покупает
            <span className="_field" mnpl-field_id="3">
              Hugo Boss
            </span>{" "}
            за 600k
          </div>
          <div className="table-body-board-chat-message" id="event_CUafxtW8AAo">
            <span
              className="table-body-board-chat-message-user _color_player_1_color _rendered"
              mnpl-userid="429935"
            >
              Константин
            </span>
            выбрасывает 3:5
          </div>
          <div className="table-body-board-chat-message" id="event_CUafxtX8AAs">
            <span
              className="table-body-board-chat-message-user _color_player_1_color _rendered"
              mnpl-userid="429935"
            >
              Константин
            </span>
            попадает на
            <span className="_field" mnpl-field_id="8">
              Puma
            </span>{" "}
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUafydq8AA4">
            <span
              className="table-body-board-chat-message-user _color_player_1_color _rendered"
              mnpl-userid="429935"
            >
              Константин
            </span>
            покупает
            <span className="_field" mnpl-field_id="8">
              Puma
            </span>{" "}
            за 1,000k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagExl8ABE">
            <span
              className="table-body-board-chat-message-user _color_player_2_color _rendered"
              mnpl-userid="578308"
            >
              Кармен
            </span>
            сдаётся
          </div>
          <div className="table-body-board-chat-message" id="event_CUagFPr8ABQ">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            выбрасывает 1:1 и получает ещё один ход, так как выпал дубль
          </div>
          <div className="table-body-board-chat-message" id="event_CUagFPr8ABU">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            попадает на поле «Шанс».
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            вспомнил, что забыл дома выключить утюг. Следующий ход будет в
            обратном направлении
          </div>
          <div className="table-body-board-chat-message" id="event_CUagGNi8ABg">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            выбрасывает 6:6
          </div>
          <div className="table-body-board-chat-message" id="event_CUagGNk8ABo">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            арестован полицией и отправляется в тюрьму
          </div>
          <div className="table-body-board-chat-message" id="event_CUagHzZ8AB0">
            <div className="table-body-board-chat-message-users">
              <a
                className="table-body-board-chat-message-user _message _color_player_3_bg_plain _rendered"
                mnpl-userid="1276527"
                href="/profile/1276527"
                target="_blank"
              >
                Влад
              </a>
            </div>
            &nbsp;—&nbsp;<span mnpl-usermessage="">...</span>
          </div>
          <div className="table-body-board-chat-message" id="event_CUagIuS8ACA">
            <span
              className="table-body-board-chat-message-user _color_player_0_color _rendered"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            выбрасывает 2:5
          </div>
          <div className="table-body-board-chat-message" id="event_CUagIuS8ACE">
            <span
              className="table-body-board-chat-message-user _color_player_0_color _rendered"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            посещает полицейский участок с экскурсией
          </div>
          <div className="table-body-board-chat-message" id="event_CUagJ2e8ACQ">
            <span
              className="table-body-board-chat-message-user _color_player_1_color _rendered"
              mnpl-userid="429935"
            >
              Константин
            </span>
            выбрасывает 2:6
          </div>
          <div className="table-body-board-chat-message" id="event_CUagJ2e8ACU">
            <span
              className="table-body-board-chat-message-user _color_player_1_color _rendered"
              mnpl-userid="429935"
            >
              Константин
            </span>
            попадает на
            <span className="_field" mnpl-field_id="16">
              Coca-Cola
            </span>{" "}
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUagJ7I8ACg">
            <div className="table-body-board-chat-message-users">
              <a
                className="table-body-board-chat-message-user _message _color_player_3_bg_plain _rendered"
                mnpl-userid="1276527"
                href="/profile/1276527"
                target="_blank"
              >
                Влад
              </a>
            </div>
            &nbsp;—&nbsp;
            <span mnpl-usermessage="">Это только я так могу</span>
          </div>
          <div className="table-body-board-chat-message" id="event_CUagK9g8ACs">
            <span
              className="table-body-board-chat-message-user _color_player_1_color _rendered"
              mnpl-userid="429935"
            >
              Константин
            </span>
            покупает
            <span className="_field" mnpl-field_id="16">
              Coca-Cola
            </span>
            за 1,800k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagLWP8AC4">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            заплатил 500k и вышел из тюрьмы
          </div>
          <div className="table-body-board-chat-message" id="event_CUagLzn8ADE">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            выбрасывает 6:3
          </div>
          <div className="table-body-board-chat-message" id="event_CUagLzn8ADI">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            попадает на
            <span className="_field" mnpl-field_id="19">
              Fanta
            </span>{" "}
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUagMa-8ADU">
            <span
              className="table-body-board-chat-message-user _color_player_3_color _rendered"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            покупает
            <span className="_field" mnpl-field_id="19">
              Fanta
            </span>{" "}
            за 2,000k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagMob8ADg">
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            выбрасывает 3:2
          </div>
          <div className="table-body-board-chat-message" id="event_CUagMob8ADk">
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            попадает на
            <span className="_field" mnpl-field_id="15">
              Audi
            </span>{" "}
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUagNL58ADw">
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            покупает
            <span className="_field" mnpl-field_id="15">
              Audi
            </span>{" "}
            за 2,000k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagRML8AD8">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            выбрасывает 1:2
          </div>
          <div className="table-body-board-chat-message" id="event_CUagRML8AEA">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            попадает на
            <span className="_field" mnpl-field_id="19">
              Fanta
            </span>{" "}
            и должен заплатить игроку
            <span
              className="table-body-board-chat-message-user _color_player_3_color"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            аренду в размере 160k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagR0Y8AEM">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            заплатил аренду 160k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagSM_8AEY">
            <span
              className="table-body-board-chat-message-user _color_player_3_color"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            выбрасывает 4:3
          </div>
          <div className="table-body-board-chat-message" id="event_CUagSM_8AEc">
            <span
              className="table-body-board-chat-message-user _color_player_3_color"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            попадает на
            <span className="_field" mnpl-field_id="26">
              McDonald's
            </span>
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUagTBf8AEo">
            <span
              className="table-body-board-chat-message-user _color_player_3_color"
              mnpl-userid="1276527"
            >
              Влад
            </span>
            покупает
            <span className="_field" mnpl-field_id="26">
              McDonald's
            </span>
            за 2,600k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagTSw8AE0">
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            выбрасывает 3:5
          </div>
          <div className="table-body-board-chat-message" id="event_CUagTSw8AE4">
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            попадает на
            <span className="_field" mnpl-field_id="23">
              Lufthansa
            </span>{" "}
            и задумывается о покупке
          </div>
          <div className="table-body-board-chat-message" id="event_CUagUIs8AFE">
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            покупает
            <span className="_field" mnpl-field_id="23">
              Lufthansa
            </span>
            за 2,200k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagUcC8AFQ">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            выбрасывает 2:2 и получает ещё один ход, так как выпал дубль
          </div>
          <div className="table-body-board-chat-message" id="event_CUagUcC8AFU">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            попадает на
            <span className="_field" mnpl-field_id="23">
              Lufthansa
            </span>{" "}
            и должен заплатить игроку
            <span
              className="table-body-board-chat-message-user _color_player_0_color"
              mnpl-userid="695820"
            >
              Гром_ка
            </span>
            аренду в размере 180k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagVMV8AFg">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            заплатил аренду 180k
          </div>
          <div className="table-body-board-chat-message" id="event_CUagVjN8AFs">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            выбрасывает 3:3 и получает ещё один ход, так как выпал дубль
          </div>
          <div className="table-body-board-chat-message" id="event_CUagVjO8AFw">
            <span
              className="table-body-board-chat-message-user _color_player_1_color"
              mnpl-userid="429935"
            >
              Константин
            </span>
            попадает на
            <span className="_field" mnpl-field_id="29">
              KFC
            </span>{" "}
            и задумывается о покупке
          </div>
        </div>
        <div className="scr-pane">
          <div
            className="scr-pane-handler"
            style={{
              transform: "translateY(284px)",
              height: "153.499px",
            }}
          ></div>
        </div>
      </div>
    </div>
    <div className="table-body-board-chatbottom">
      <div className="_channel" mnpl-ch="all"></div>
      <div className="_input">
        <input placeholder="Введите сообщение" />
        <div
          mnpl-emote="Clap"
          className="emotes-button"
          style={{
            backgroundImage: `url("//m1.dogecdn.wtf/chat/eClap.svg")`,
          }}
        ></div>
        <div className="_hint _hint_tab">
          Теперь можно писать конкретному игроку!
          <br />
          <span></span>, чтобы выбрать получателя.
        </div>
      </div>
      <div className="_hidekeyboard"></div>
      <div className="_screenshot" style={{ display: "none" }}></div>
      <div className="_fullscreen"></div>
      <div className="_fullscreenoff" style={{ display: "none" }}></div>
      <div className="_helper"></div>
    </div>
  </>
);
