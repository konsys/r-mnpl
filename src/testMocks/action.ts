import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
