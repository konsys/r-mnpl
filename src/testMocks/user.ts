import { IPlayer, IUser } from "types/types";
import { PlayerRoomStatus, RoomPlayer } from "stores/Game/Rooms/RoomsModel";

export const testPlayer1: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "IVAN",
  team: undefined,
  avatar: "testAvatar1",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: true,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testPlayer2: IPlayer = {
  frags: "",
  userId: 1,
  money: 15000,
  vip: true,
  registrationType: "none",
  name: "DIMAN",
  team: undefined,
  avatar: "testAvatar2",
  createdAt: new Date(),
  updatedAt: new Date(),
  isActive: false,
  isBlocked: true,
  isActing: false,
  gameId: "",
  doublesRolledAsCombo: 0,
  jailed: 0,
  unjailAttempts: 0,
  meanPosition: 0,
  creditPayRound: false,
  creditNextTakeRound: 0,
  score: 0,
  additionalTime: 0,
  timeReduceLevel: 0,
  creditToPay: 0,
  canUseCredit: false,
  moveOrder: 0,
};

export const testUser: IUser = {
  userId: 1,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const test5User: IUser = {
  userId: 5,
  vip: false,
  name: "NIKOLA",
  isActive: true,
  isBlocked: false,
};
export const testAvatarUser: IUser = {
  avatar: "/test/avatar.jpg",
  userId: 2,
  vip: false,
  name: "IVAN",
  isActive: true,
  isBlocked: false,
};
export const testVipUser: IUser = {
  userId: 3,
  vip: true,
  name: "OLEG",
  isActive: true,
  isBlocked: false,
};

export const testRoomUser: RoomPlayer = {
  playerRoomStatus: PlayerRoomStatus.ACITVE,
  userId: 4,
  vip: true,
  name: "MISHA",
  isActive: true,
  isBlocked: false,
};
