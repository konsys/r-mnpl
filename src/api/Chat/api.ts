import { IChatMessage } from "components/core/Game/GameChat/ChatMessage";
import { client } from "http/client";

export const getChatMessages = async (): Promise<IChatMessage[]> =>
  await (await client.get(`/chat`)).data;
