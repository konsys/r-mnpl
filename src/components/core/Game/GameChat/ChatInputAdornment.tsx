import { deleteReplyToEvent, repliesTo$ } from "stores/Game/Chat/GameChatModel";

import { InputAdornment } from "@material-ui/core";
import PlayerChip from "./PlayerChip";
import React from "react";
import { useStore } from "effector-react";

export default function ChatInputAdornment() {
  const replyTo = useStore(repliesTo$);
  return (
    <div>
      <InputAdornment position="start" variant="outlined">
        {replyTo && replyTo.users
          ? replyTo.users.map((v, k) => (
              // TODO make standart spacing
              <div style={{ marginRight: "5px" }} key={k}>
                <PlayerChip
                  key={k}
                  handleDelete={() => deleteReplyToEvent(v)}
                  name={v.name}
                />
              </div>
            ))
          : ""}
      </InputAdornment>
    </div>
  );
}
