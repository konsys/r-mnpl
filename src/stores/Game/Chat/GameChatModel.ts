import { GameDomain } from "stores/Board/BoardDomain";
import { IChatMessage } from "components/core/Game/GameChat/ChatMessage";
import { IUser } from "types/types";
import { createGate } from "effector-react";
import { postChat } from "api/GameChat/api";
import { getChatMessages } from "api/Chat/api";
import { merge } from "lodash";

const ChatDomain = GameDomain.domain("ChatDomain");

export interface IChatMessageRequest {
  message: string;
  replies: IUser[];
}

export const chatGate = createGate();

export const setChatMessages = ChatDomain.event<IChatMessage[]>();
export const resetChatMessages = ChatDomain.event();

export const sendChatMessageFx = ChatDomain.effect<
  IChatMessageRequest,
  IChatMessage[]
>({
  handler: postChat,
});

export const getChatMessagesFx = ChatDomain.effect<void, IChatMessage[]>({
  name: "getChatMessagesFx",
  handler: getChatMessages,
});

export const gameChat$ = ChatDomain.store<IChatMessage[]>([])
  .on(setChatMessages, (_, v) => v)
  .on(getChatMessagesFx.done, (_, { result }) => result)
  .reset(merge([chatGate.close, resetChatMessages]));

export const addReplyToChatMessage = ChatDomain.event<IUser>();
export const deleteReplyToChatMessage = ChatDomain.event<IUser>();
export const resetReplyToChatMessage = ChatDomain.event();

export interface IReply {
  n: number;
  users: IUser[];
}
export const repliesToChatMessage$ = ChatDomain.store<IReply>({
  n: 0,
  users: [],
})
  .on(addReplyToChatMessage, (prev, v) => {
    const users = prev.users.filter((user) => user.userId !== v.userId);
    users.push(v);
    return { n: users.length, users };
  })
  .on(deleteReplyToChatMessage, (prev, v) => {
    const r = prev ? prev.users.filter((user) => user.userId !== v.userId) : [];
    return {
      n: r ? r.length : 0,
      users: r ? r : [],
    };
  })
  .reset(resetReplyToChatMessage);

chatGate.open.watch(() => getChatMessagesFx());
