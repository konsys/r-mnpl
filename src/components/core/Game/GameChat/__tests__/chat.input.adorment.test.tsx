import * as chat from "stores/Game/Chat/GameChatModel";

import ChatInputAdornment from "../ChatInputAdornment";
import PlayerChip from "../PlayerChip";
import React from "react";
import { addReplyToEvent } from "stores/Game/Chat/GameChatModel";
import { shallow } from "enzyme";
import { testUser } from "testMocks/user";

describe("test input adorment", () => {
  it("should render", () => {
    expect(shallow(<ChatInputAdornment />)).toMatchSnapshot();
  });
  it("shouldn't render replies", () => {
    expect(shallow(<ChatInputAdornment />).find(PlayerChip)).toHaveLength(0);
  });

  it("shouldn't render replies", () => {
    addReplyToEvent(testUser);
    expect(shallow(<ChatInputAdornment />).find(PlayerChip)).toHaveLength(1);
  });

  it("should render replies", () => {
    addReplyToEvent(testUser);
    expect(
      shallow(<ChatInputAdornment />)
        .find(PlayerChip)
        .get(0).props.name
    ).toBe(testUser.name);
  });

  it("should delete chip", () => {
    addReplyToEvent(testUser);
    const testMock = chat as any;
    const deleteMock = jest.spyOn(testMock, "deleteReplyToEvent");
    shallow(<ChatInputAdornment />)
      .find(PlayerChip)
      .get(0)
      .props.handleDelete();

    expect(deleteMock).toHaveBeenCalledTimes(1);
  });
});
