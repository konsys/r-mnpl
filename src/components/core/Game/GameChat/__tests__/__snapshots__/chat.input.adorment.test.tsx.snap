// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test input adorment should render 1`] = `ShallowWrapper {}`;
